### ISSUE TITLE
Coverage report is not correct when Rollup Inject is used

### ISSUE THREAD

**Issue by gian1200** [modernweb-dev/web#2778]
I'm currently migrating from Karma to Web-test-runner. When using [Rollup Inject](https://github.com/rollup/plugins/tree/master/packages/inject), some files get modified to include additional `import` lines. This additions are not taking into consideration when generating the coverage report. The result of this is that covered/uncovered lines are not correct (offset depending on the amount of lines added).

#### Karma (correct):
![image](https://github.com/user-attachments/assets/00371c82-8bea-48ab-a313-6505a7ea1503)

#### Web-test-runner:
![image](https://github.com/user-attachments/assets/6e445c64-8852-4590-82c8-fd30a5316771)

**Is this a bug, or is there any other plugin that may fix this issue?**

Current plugins:
```js
...
"plugins": [
	legacy({
		"node_modules/jquery/dist/jquery.js": "$"
	}),
	inject({
		"$": "jquery",
		"fetchMock": "fetch-mock/esm/client",
		"Chart": ["chart.js/auto", "Chart"]
	})
]
...
```

**Comment by gian1200** [comment]
No sure if I got your point. Are you asking if the 2 offset lines correlate with the amount of lines added by `inject`? If that's the question, then yes, they match.

Regarding the `source maps`, where/how can I validate it? With that information maybe I can report it to `@rollup/plugin-inject`.

I reported it here first because [the docs](https://modern-web.dev/docs/dev-server/plugins/rollup/#compatibility-with-rollup-plugins) stated that the plugin was "tested to work correctly".

Could it also be an issue with `@web/dev-server-rollup`?

**Comment by Westbrook** [comment]
These plugins are tested to work correctly with Web Dev Server and there is some assumption that this also means that they work with Test Runner and _all_ of its associated options, but clearly something was missed or changed over time. So, thanks for bringing this up!

The lines being exactly the lines added by inject implies that it will be an issue with that plugin; possibly on their end, possibly on ours. It would be good to check with them to see if their changes are passed into source map tooling that would be needed to ensure the lines match up from the actual code under test vs the source.

**Comment by gian1200** [comment]
I tried to build a small reproducer with `rollup` only.

TLDR: generated source map seems to be OK

----

![image](https://github.com/user-attachments/assets/59d9886f-d329-4b86-9711-d59508af405f)

- `src/index.mjs`:
```js
$("");
```

- `rollup.config.mjs`:
```js
import inject from '@rollup/plugin-inject';

/**
 * @type {import("rollup").RollupOptions}
 */
export default {
	input: 'src/index.mjs',
	output: {
		sourcemap: true,
		dir: 'output'
		// format: 'cjs'
	},

	plugins: [
		inject({
			"$": "jquery"
			// Promise: ['es6-promise', 'Promise']
		})
	]
};

```

- `output/index.js`:
```js
import $ from 'jquery';

$("");
//# sourceMappingURL=index.js.map
```

- `output/index.js.map`:
```js
{
	"version": 3,
	"file": "index.js",
	"sources": [
		"../src/index.mjs"
	],
	"sourcesContent": [
		"$(\"\");"
	],
	"names": [],
	"mappings": ";;AAAA,CAAC,CAAC,EAAE,CAAC"
}
```

Used a [source-map-visualization](https://evanw.github.io/source-map-visualization/) to validate the source map generated and got the following:

![image](https://github.com/user-attachments/assets/9f18aea1-c6ae-493d-a07a-1551d3dcd77e)

https://evanw.github.io/source-map-visualization/#MAAxODMAew0KCSJ2ZXJzaW9uIjogMywNCgkiZmlsZSI6ICJpbmRleC5qcyIsDQoJInNvdXJjZXMiOiBbDQoJCSIuLi9zcmMvaW5kZXgubWpzIg0KCV0sDQoJInNvdXJjZXNDb250ZW50IjogWw0KCQkiJChcIlwiKTsiDQoJXSwNCgkibmFtZXMiOiBbXSwNCgkibWFwcGluZ3MiOiAiOztBQUFBLENBQUMsQ0FBQyxFQUFFLENBQUMiDQp9
